docs/Deprecating-Disabling-and-Removing-Formulae.md

```diff
@@ -0,0 +1,110 @@
+
+**Note: disabled formulae in homebrew/core will be automatically removed one year after their disable date**
+
+To disable a formula, add a `disable!` call. This call should include a deprecation date (in the ISO 8601 format) and a deprecation reason:
+
+```ruby
+disable! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the reason for disabling came into effect. If there is no clear date but the formula needs to be disabled, use today's date. If the `date` parameter is set to a date in the future, the formula will be deprecated until that date (on which the formula will become disabled).
+
+The `because` parameter can be set to a preset reason (using a symbol) or a custom reason. See the [Deprecate and Disable Reasons](#deprecate-and-disable-reasons) section below for more details about the `because` parameter.
+
+## Removal
+
+A formula should be removed if there is a serious issue with the formula or the formula has been disabled for a long period of time.
+
+A formula should be removed if it has been disabled for a long period of time, it has a non-open-source license, or there is another serious issue with the formula that makes it not compatible with homebrew/core.
+
+**Note: disabled formulae in homebrew/core will be automatically removed one year after their disable date**
+
+## Deprecate and Disable Reasons
+
+When a formula is deprecated or disabled, a reason explaining the action should be provided.
+
+There are two ways to indicate the reason. The preferred way is to use a pre-existing symbol to indicate the reason. The available symbols are:
+
+- `:does_not_build`: the formulae that cannot be build from source
+- `:no_license`: the formulae does not have a license
+- `:repo_archived`: the upstream repository has been archived
+- `:repo_removed`: the upstream repository has been removed
+- `:unmaintained`: the project appears to be abandoned
+- `:unsupported`: Homebrew's application of the software is not supported by the upstream developers (e.g. upstream only supports macOS versions prior to 10.14)
+- `:deprecated_upstream`: the project is deprecated upstream
+- `:versioned_formula`: the formula is a versioned formula
+
+These reasons can be specified by their symbols (the comments show the message that will be displayed to users):
+
+```ruby
+# Warning: <formula> has been deprecated because it is deprecated upstream!
+deprecate! date: "2020-01-01", because: :deprecated_upstream
+```
+
+```ruby
+# Error: <formula> has been disabled because it does not build!
+disable! date: "2020-01-01", because: :does_not_build
+```
+
+If these pre-existing reasons do not fit, a custom reason can be specified. These reasons should be written to fit into the sentence `<formula> has been deprecated/disabled because it <reason>!`.
+
+Here are examples of a well-worded custom reason:
```

Comment:
I'm not sure we have to teach people grammar in the homebrew documentation.

Reply:
I agree this may be too much handholding but we do have a few issues currently in homebrew/core so I don't think this part should totally be removed.

See [`liblwgeom`](https://github.com/Homebrew/homebrew-core/blob/3205250b454f50e27138b94f515aeda96dc73b06/Formula/liblwgeom.rb#L21):

```
Warning: liblwgeom has been deprecated because it liblwgeom headers are not installed anymore, use librttopo instead!
```

And [`exomizer`](https://github.com/Homebrew/homebrew-core/blob/3205250b454f50e27138b94f515aeda96dc73b06/Formula/exomizer.rb#L22):

```
Error: exomizer has been disabled because it custom license!
```

Reply:
I think a formula author/maintainer would reasonably expect:
```
  ... because: "<reason>"
```
to produce the message:
```
  [This is happening] because <reason>.
```
rather than:
```
  [This is happening] because it <reason>!
```
In the case of `exomizer`, "because it custom license" is just _wrong_, but "because custom license" accords with modern usage, though it's anathema for grammar pedants. :laughing:

Reply:
@gromgit I agree that this makes sense. This is how I initially created it, but at some point, it was decided that since most of the reasons already started with "it", the "it" should be added by default. See https://github.com/Homebrew/brew/pull/8549#issuecomment-683924407 and https://github.com/Homebrew/homebrew-core/pull/60321#issuecomment-682642693. TBH I think it could be changed, now that we have the symbols available.